lstchain/calib/camera/utils.py

```diff
@@ -0,0 +1,62 @@
+import tables
+import numpy as np
+
+from lstchain.io.io import dl1_params_tel_mon_ped_key, dl1_params_tel_mon_cal_key
+
+
+def get_bias_and_std(dl1_file):
+    """
+    Function to extract bias and std of pedestal from interleaved events from dl1 file.
+    Parameters
+    ----------
+    input_filename: str
+        path to dl1 file
+    Returns
+    -------
+    bias, std: np.ndarray, np.ndarray
+        bias and std in p.e.
+    """
+    f = tables.open_file(dl1_file)
+    ped = f.root[dl1_params_tel_mon_ped_key]
+    ped_charge_mean = np.array(ped.cols.charge_mean)
+    ped_charge_std = np.array(ped.cols.charge_std)
+    calib = f.root[dl1_params_tel_mon_cal_key]
+    dc_to_pe = np.array(calib.cols.dc_to_pe)
+    ped_charge_mean_pe = ped_charge_mean * dc_to_pe
+    ped_charge_std_pe = ped_charge_std * dc_to_pe
+    f.close()
+    return ped_charge_mean_pe, ped_charge_std_pe
+
+def get_threshold_from_dl1_file(dl1_path, sigma_clean):
+    """
+    Function to get picture threshold from dl1 from interleaved pedestal events.
+    Parameters
+    ----------
+    input_filename: str
+        path to dl1 file
+    sigma_clean: float
+        cleaning level
+    Returns
+    -------
+    picture_thresh: np.ndarray
+        picture threshold calculated using interleaved pedestal events
+    """
+    high_gain = 0
+    ped_mean_pe, ped_std_pe = get_bias_and_std(dl1_path)
+
+    # If problem with interleaved pedestal std values occur, take pedestal
+    # std values from calibration run.
+    # Correct interleaved pedestal std array should have shape (2,2,1855)
+    if ped_std_pe.shape[0] == 2:
+        interleaved_events_id = 1
+    else:
+        interleaved_events_id = 0
+    threshold_clean_pe = ped_mean_pe + sigma_clean*ped_std_pe
+    # find pixels with std = 0 and mean = 0 <=> dead pixels in interleaved
+    # pedestal event likely due to stars
+    dead_pixel_ids = np.where(threshold_clean_pe[interleaved_events_id, high_gain, :] == 0)[0]
+    # for dead pixels set max value of threshold
```

Comment:
Why the max in any pixel? Seems a bit arbitrary - but irrelevant if pixel is really switched off.

Reply:
If we have strong star in camera FoV some pixels are switched off (std of signal in this pixels is 0) 
![image](https://user-images.githubusercontent.com/8089538/106906405-6d36d080-66fd-11eb-8199-05c64dbb83ca.png) , so for those pixels, I set max value from another pixels.

Reply:
Yes, it just seemed a bit arbitrary to set the max of all pixels, but no problem. I don't have a better suggestion.

Reply:
It might have been a better solution to simply use 0 for those pixels. This means that they would have the standard image cleaning applied to them, which as Abelardo says does not matter since they have no signal.

Reply:
> It might have been a better solution to simply use 0 for those pixels. This means that they would have the standard image cleaning applied to them, which as Abelardo says does not matter since they have no signal.

OK, let me test this idea.

Reply:
In this case (use 0 for those pixels), number of surviving pedestal images increased a bit from 48 to 56 (with 224 pedestal images in total)

Reply:
> In this case (use 0 for those pixels), number of surviving pedestal images increased a bit from 48 to 56 (with 224 pedestal images in total)

So despite they have 0 mean pedestal and std dev they are not dead, right? Perhaps the mean and std are 0 because they did not reach a minimum number of valid events for the calculation. It may be safer then to keep a high threshold.  
By the way the fraction of pedestal survivors is pretty high, are you testing with low thresholds?

Reply:
I was testing Run 3004 with (picture threshold = 8, boundary th. = 4 and sigma=3), this run is noisy. I will try find better value of threshold.

Reply:
what about using a value in an invalid range so those pixels are ruled out? We do not want them surviving any cleaning.